babel/numbers.py

```diff
@@ -624,6 +625,35 @@ def _format_currency_long_name(
     return unit_pattern.format(number_part, display_name)
 
 
+def format_compact_currency(number, currency, *, format_type="short", locale=LC_NUMERIC, fraction_digits=0):
+    u"""Format a number as a currency value in compact form.
+
+    >>> format_compact_currency(12345, 'USD', locale='en_US')
+    u'$12K'
+    >>> format_compact_currency(123456789, 'USD', locale='en_US', fraction_digits=2)
+    u'$123.46M'
+    >>> format_compact_currency(123456789, 'EUR', locale='de_DE', fraction_digits=1)
+    u"123,5\xa0Mio'.'\xa0€"
```

Comment:
This is actually incorrectly formatted – using `print()` to clean out the NBSPs:
```
>>> print(u"123,5\xa0Mio'.'\xa0€")
123,5 Mio'.' €
```
The output should be
```
123,5 Mio. €
```
i.e. the literal segment `'.'` should get correctly interpreted.

Good choice of doctest case 😁